torch/csrc/Generator.cpp

```diff
@@ -52,12 +56,20 @@ static PyObject* THPGenerator_pynew(
   auto device = r.deviceWithDefault(0, at::Device(at::kCPU));
 
   THPGeneratorPtr self((THPGenerator*)type->tp_alloc(type, 0));
-#ifdef USE_CUDA
+#if defined(USE_CUDA) || defined(USE_MPS)
```

Comment:
This logic is a bit weird tbh.
Let's just do:
```
if (device.type() == at::kCPU) {
    self->cdata = make_generator<CPUGeneratorImpl>();
}
#ifdef USE_CUDA
  else if (device.type() == at::kCUDA) {
    self->cdata = make_generator<CUDAGeneratorImpl>(device.index());
  }
#endif
#if USE_MPS
  else if (device.type() == at::kMPS) {
    self->cdata = make_generator<MPSGeneratorImpl>();
  }
#endif
else {
  TORCH_CHECK(false, ...);
}
```

Reply:
This was based on the original weird CUDA logic which creates "CPU" instance in a `#ifdef USE_CUDA` block:
```
#ifdef USE_CUDA
  if (device.type() == at::kCPU) {
    self->cdata = make_generator<CPUGeneratorImpl>();
  } else if (device.type() == at::kCUDA) { 
```
I didn't want to modify the original code, so I just married MPS into it :)
I'll try your suggestion.

Reply:
Feel free to cleanup weird cuda code ;)

Reply:
Updated the PR.